### DISCUSSION TITLE
Missing Controller Firmware Update Selection

### DISCUSSION THREAD

**Discussion by AseKarlsson** [zwave-js/zwave-js-ui#3010] [General]
Hi,
I earlier found/used the UI to update Controller FIrmware thru ZWave-JS UI on HomeAssistant.
Now there was a new firmware released to my Controller, but I cannot find this update feature anymore, has it been removed?
I am on the latest version found by HA.

Thanks,
Ase

**Comment by robertsLando** [comment]
Control Panel > Press on the fab > Advanced actions > OTW update.

It has been moved there because if there are problems the controller node could be not visible so if it is on expanded node advanced actions users would not be able to trigger it

**Reply by AseKarlsson** [reply]
Thanks Robert,
I think I am in the right spot then, but cannot see it. Can find it there for other devices.
Think it was here before, last time I used it.

![image](https://user-images.githubusercontent.com/80900477/226115749-f7404857-78a9-4c60-9250-814614ff3f7c.png)

**Reply by kpine** [reply]
> Control Panel > Press on the fab > Advanced actions > OTW update.

![image](https://user-images.githubusercontent.com/791794/226145488-d5a0746a-42e1-4432-97e2-bfcbbd8cc35d.png)

**Reply by AseKarlsson** [reply]
Noep, missing for me. Lets hope next update will solve this

**Reply by kpine** [reply]
Can you confirm which version you are running? Do you have a screenshot of what the advanced actions looks like for you?

**Reply by AseKarlsson** [reply]
Here is my version:
![image](https://user-images.githubusercontent.com/80900477/226186057-2793c406-b7e1-43f9-bb7e-97ba894ad926.png)

After selecting "Advanced" for the Controller I get the following options:
![image](https://user-images.githubusercontent.com/80900477/226186096-93cb0f90-e3e9-40b2-873a-835986482d45.png)

If i Select a Plug I get:
![image](https://user-images.githubusercontent.com/80900477/226186173-a918f13b-0cd1-41aa-a40a-edb74446e87c.png)

**Reply by kpine** [reply]
Hi, you seem to be missing the instructions. Take a look at my screenshot again. Advanced Actions is from the main page. As was already mentioned, the OTW Update was moved out of the node settings.

**Reply by kpine** [reply]
![image](https://user-images.githubusercontent.com/791794/226186377-f0c31c08-d2fe-4318-9cbd-11fadc196fe3.png)

**Reply by AseKarlsson** [reply]
Hi,
Thanks for the PrintScreen as clarification, now there is no missunderstanding left.

I am missing the Actions Tab all together
![image](https://user-images.githubusercontent.com/80900477/226186839-bc667739-32c7-4662-b116-b46df2543191.png)

Should that be enabled somewhere?

**Reply by kpine** [reply]
Yes, you are missing the Actions button which is pretty strange. Can you try refreshing the browser (e.g. F5), and then also opening it in a private/incognito window and see if there's any difference?

**Reply by kpine** [reply]
I just upgraded to 8.11.0 from 8.9.0. 😅 @robertsLando 

![image](https://user-images.githubusercontent.com/791794/226189795-302cffcb-5a6e-49f2-b484-16f4d3428be7.png)

**Reply by AseKarlsson** [reply]
Thanks @kpine for being so helpful!
Good to see that I am not by myself with this problem, lets hope for a fix!

**Reply by kpine** [reply]
Also tried 8.10.1, and the button is missing there as well. Rolled back to 8.9.0 and it's back.

**Reply by robertsLando** [reply]
The button has been moved to bottom right corner, it's a fab with a plus icon, press it and it wull show you the actions

**Reply by AseKarlsson** [reply]
Thanks @robertsLando for this info, found it now! (I think it is better placed under the Controller/Advanced)

**Reply by robertsLando** [reply]
@AseKarlsson I understand but as said if you start the application in boot loader only mode the controller node would not be visible, this mode is useful to restore from problems related to failed otw updates so placing it there allows users to do it even without nodes in the table

**Reply by AseKarlsson** [reply]
Could it be in in both places for the best for all (not an issue, now when I know where to find it), just as an idea.
Thanks!

**Reply by kpine** [reply]
> The button has been moved to bottom right corner, it's a fab with a plus icon, press it and it wull show you the actions

OK, thanks for the clarification. I think the new approach is hard to discover, but that's just my personal opinion, I don't particularly care how you do it anyways. It would be nice however to see big UI changes like this specifically called out in the release notes. It's not obvious from the list of PRs in the notes which change this landed in, unless I missed something obvious.

**Reply by AseKarlsson** [reply]
@robertsLando, Just curious, what is "fab" shorten of (not native English speaking), this did throe me off in your first reply. 
For me it would be button with + sign on it?

**Reply by kpine** [reply]
"Fab" seems to be an acronym for "Floating Action Button". Example: https://developer.android.com/develop/ui/views/components/floating-action-button. So, yes, it's the blue + button.

**Reply by robertsLando** [reply]
@kpine I'm sorry for that, it has been shipped with the general ui improvements for smartphones, the old way could be better with large screens but not so in smaller ones, I could put it back BTW and show it only on larger screens. I thought about that but FAB is something largely used in most UI design and didn't think it was so hard to discover, but I agree that if someone used to see it on top it's not so easy at first sight... I'm open to new ideas BTW :) 

@AseKarlsson FAB stands for floating action button, it's a material design component

**Reply by kpine** [reply]
No worries, it's fine by me. 👍 Just takes a minute to adjust.
